docs/new-resource.md

```diff
@@ -1,149 +1,217 @@
+	return &EC2InstanceEnumerator{
+		repository: repo,
+		factory:    factory,
 	}
 }
-```
 
-Then in the cloud provider's init file (e.g. in `remote/aws/init.go`), add your new implementation for `resource.Supplier`:
+func (e *EC2InstanceEnumerator) SupportedType() resource.ResourceType {
+	return aws.AwsInstanceResourceType
+}
 
-```go
-func Init() error {
-	provider, err := NewTerraFormProvider()
+func (e *EC2InstanceEnumerator) Enumerate() ([]*resource.Resource, error) {
+	instances, err := e.repository.ListAllInstances()
 	if err != nil {
-		return err
+		return nil, remoteerror.NewResourceListingError(err, string(e.SupportedType()))
+	}
+
+	results := make([]*resource.Resource, len(instances))
+
+	for _, instance := range instances {
+		results = append(
+			results,
+			e.factory.CreateAbstractResource(
+				string(e.SupportedType()),
+				*instance.InstanceId,
+				map[string]interface{}{},
+			),
+		)
 	}
 
-	terraform.AddProvider(terraform.AWS, provider)
-	resource.AddSupplier(NewS3BucketSupplier(provider.Runner().SubRunner(), s3.New(provider.session)))
-	...
+	return results, err
 }
 ```
 
-Don't forget to add unit tests after adding a new resource.
-You can also add acceptance tests if you think it makes sense.
+As you can see, listing error are treated in a particular way. Instead of failing and stopping the scan they will be handled, and an alert will be created.
+So please don't forget to wrap these errors inside a `NewResourceListingError`.
+For some provider error handling is not that coherent, so you might need to check in `pkg/remote/resource_enumeration_error_handler.go` and add a new case for your error.
+You should test enumerator behavior when you do not have permission to enumerate resource, in the snippet above, `ListAllInstances` may return an `AccessDenied` error or so that should be handled.
+
+Once the enumerator is written you have to add it to the remote init located in `pkg/remote/<providername>/init.go` :
+```go
+    remoteLibrary.AddEnumerator(NewEC2InstanceEnumerator(s3Repository, factory))
+```
+
+### DetailsFetcher
+
+DetailsFetcher are only used by driftctl experimental deep mode.
+
+This is the component that call terraform provider to retrieve the full attribute for each resource.
+We do not want to reimplement what has already been done in every terraform provider, so you should not call the remote sdk there.
+
+If `common.GenericDetailsFetcher` satisfy your needs you should always prefer using it instead of implementing a custom `DetailsFetcher` in a new struct.
+
+The `DetailsFetcher` should also be added to `pkg/remote/<providername>/init.go` even if you use the generic version :
+```go
+    remoteLibrary.AddDetailsFetcher(aws.AwsEbsVolumeResourceType, common.NewGenericDetailsFetcher(aws.AwsEbsVolumeResourceType, provider, deserializer))
+```
+
+
+***Don't forget to add unit tests after adding a new resource.***
+
+You can find example of "functional" tests in pkg/remote/<type>_scanner_test.go
```

Comment:
Backticks around the file and the `.` at the end of the sentence.

Reply:
I would rather put emphasis with bold text for **functional tests** rather than simple "